x-pack/plugins/enterprise_search/public/applications/app_search/components/document_creation/document_creation_logic.test.ts

```diff
@@ -163,5 +195,393 @@ describe('DocumentCreationLogic', () => {
+            documents: [mockJson],
+          });
+          expect(DocumentCreationLogic.actions.setErrors).not.toHaveBeenCalled();
+        });
+      });
+    });
+
+    describe('uploadDocuments', () => {
+      describe('valid uploads', () => {
+        const mockValidDocuments = [{ foo: 'bar', bar: 'baz', qux: 'quux' }];
+        const mockValidResponse = {
+          errors: [],
+          validDocuments: { total: 3, examples: mockValidDocuments },
+          invalidDocuments: { total: 0, examples: [] },
+          newSchemaFields: ['foo', 'bar', 'qux'],
+        };
+
+        beforeAll(() => {
+          mount();
+          jest.spyOn(DocumentCreationLogic.actions, 'setSummary');
+          jest.spyOn(DocumentCreationLogic.actions, 'setCreationStep');
+        });
+
+        it('should set and show summary from the returned response', async () => {
+          const { http } = HttpLogic.values;
+          const promise = (http.post as jest.Mock).mockReturnValueOnce(
+            Promise.resolve(mockValidResponse)
+          );
+
+          await DocumentCreationLogic.actions.uploadDocuments({ documents: mockValidDocuments });
+          await promise;
+
+          expect(DocumentCreationLogic.actions.setSummary).toHaveBeenCalledWith(mockValidResponse);
+          expect(DocumentCreationLogic.actions.setCreationStep).toHaveBeenCalledWith(
+            DocumentCreationStep.ShowSummary
+          );
+        });
+      });
+
+      describe('invalid uploads', () => {
+        beforeAll(() => {
+          mount();
+          jest.spyOn(DocumentCreationLogic.actions, 'setErrors');
+        });
+
+        it('handles API errors', async () => {
+          const { http } = HttpLogic.values;
+          const promise = (http.post as jest.Mock).mockReturnValueOnce(
+            Promise.reject({
+              body: {
+                statusCode: 400,
+                error: 'Bad Request',
+                message: 'Invalid request payload JSON format',
+              },
+            })
+          );
+
+          await DocumentCreationLogic.actions.uploadDocuments({ documents: [{}] });
+          await promise;
+
+          expect(DocumentCreationLogic.actions.setErrors).toHaveBeenCalledWith(
+            '[400 Bad Request] Invalid request payload JSON format'
+          );
+        });
+
+        it('handles client-side errors', async () => {
+          const { http } = HttpLogic.values;
+          const promise = (http.post as jest.Mock).mockReturnValueOnce(new Error());
+
+          await DocumentCreationLogic.actions.uploadDocuments({ documents: [{}] });
+          await promise;
+
+          expect(DocumentCreationLogic.actions.setErrors).toHaveBeenCalledWith(
+            "Cannot read property 'total' of undefined"
+          );
+        });
+
+        // NOTE: I can't seem to reproduce this in a production setting.
+        it('handles errors returned from the API', async () => {
```

Comment:
I'm not sure if I should keep this comment or not, I haven't been able to actually QA/repro this functionality (getting a `summary.errors` response) in production, but our documents endpoint does seem to indicate it's there (if I'm reading the Ruby code correctly, which it's very possible I'm not).

That being said there's other scenarios that realistically wouldn't happen that we're still guarding for (such as a non-file or corrupted file being submitted) but not actively QAing for, so if that's one of those scenarios maybe I should just not leave the comment because it's not super helpful.

Reply:
I don't have an opinion here.